### PR TITLE
Fixes S2 bounding volumes

### PR DESCRIPTION
**Pull request by sanjeetsuhag**
This PR fixes some two main issues in the current S2 implementation:
1. Incorrect conversion from `{level}/{x}/{y}/{z}` to Hilbert/S2 index.
2. The parent bounding volume being updated when a child bounding volume is created.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @sanjeetsuhag!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/s2-debug/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Core/S2Cell.js [hunk 1]
```diff
@@ -419,6 +419,50 @@ S2Cell.prototype.getVertex = function (index, ellipsoid) {
...
+  ).join("0");
+  var positionSuffixPadding = Array(S2_POSITION_BITS - 2 * level).join("0");
+
+  // eslint-disable-next-line
+  var cellId = BigInt(
+    "0b" +
+      faceBitString +
+      positionPrefixPadding +
+      positionBitString +
+      "1" +
```
Comment (ptrgags) [inline Source/Core/S2Cell.js]: comment that this is the sentinel bit.

File: Source/Scene/TileBoundingS2Cell.js [hunk 2]
```diff
@@ -57,6 +56,7 @@ function TileBoundingS2Cell(options) {
     ellipsoid
   );
   this._boundingPlanes = boundingPlanes;
+  this._token = options.token;
```
Comment (ptrgags) [inline Source/Scene/TileBoundingS2Cell.js]: remove this if it's not used.

**Review by ptrgags — COMMENTED** [review]
Looks good, my only feedback is a couple documentation updates.

**Comment by ptrgags** [comment]
@sanjeetsuhag I'll merge this once CI passes

**Comment by ptrgags** [comment]
merged! Thanks @sanjeetsuhag!

**Pull request merged**